### PR TITLE
PBI-5336 React Native message observer cleanup

### PR DESCRIPTION
**Pull request by ChaseApptentive**
[PBI-5336]
Cleaned up the messageCenterNotificaitonObservable in onDestroy. Should prevent memory leaks (if there are any)

[PBI-5336]: https://apptentive.atlassian.net/browse/PBI-5336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt [hunk 1]
```diff
@@ -67,20 +68,23 @@ class ApptentiveModule(private val reactContext: ReactApplicationContext) :
...
-    Apptentive.messageCenterNotificationObservable.observe { notification ->
-      Log.v(REACT_NATIVE_TAG, "Message Center notification received: $notification")
-      val eventEmitter =
-        reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-      val map = WritableNativeMap()
-      map.putInt("count", notification?.unreadMessageCount ?: 0)
-
-      eventEmitter.emit(UNREAD_MESSAGE_COUNT_CHANGED, map)
-    }
+    Apptentive.messageCenterNotificationObservable.observe(::observeNewMessage)
```
Comment (PoornimaApptentive) [inline android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt:71]: ReactNative doesn't observe for SurveyFinished events through `eventNotificationObservable`?

Comment (PoornimaApptentive) [reply android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt:71]: Also, There is no control over if the app wants to really listen to the events? Flutter exposes a method called `registerListeners`. Only if this is called from the app the listeners are registered.

Comment (ChaseApptentive) [reply android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt:71]: Not currently.

No, it is always transmitting whether it is listened to or not.

**Review by PoornimaApptentive — APPROVED** [review]

File: android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt (hunk 1 above)
Comment (PoornimaApptentive) [reply android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt:71]: In native SDK we expose that API to the app to make its choice. Is there a reason, you chose this way?

**Pull request merged**